nixos/modules/system/boot/loader/limine/limine.nix

```diff
@@ -0,0 +1,210 @@
+      example = literalExpression ''
+        { "efi/memtest86/memtest86.efi" = "''${pkgs.memtest86-efi}/BOOTX64.efi"; }
+      '';
+      description = ''
+        A set of files to be copied to {file}`/boot`. Each attribute name denotes the
+        destination file name in {file}`/boot`, while the corresponding attribute value
+        specifies the source file.
+      '';
+    };
+
+    useStorePaths = mkOption {
+      default = false;
+      type = types.bool;
+      description = ''
+        Use direct Nix store paths instead of copying boot files onto the boot partition. This is
+        noticeably slower but can greatly reduce the space usage on the boot partition.
+      '';
+    };
+
+    validateChecksums = mkOption {
+      default = true;
+      type = types.bool;
+      description = ''
+        Whether to validate file checksums before booting.
+      '';
+    };
+
+    panicOnChecksumMismatch = mkOption {
+      default = false;
+      type = types.bool;
+      description = ''
+        Whether or not checksum validation failure should be a fatal
+        error at boot time.
+      '';
+    };
+
+    package = mkOption {
+      default = pkgs.limine;
+      defaultText = literalExpression "pkgs.limine";
+      type = types.package;
+      description = ''
+        Which Limine package to use for installation.
+      '';
+    };
+
+    installEfi = mkOption {
+      default = pkgs.stdenv.hostPlatform.isEfi;
+      defaultText = literalExpression "true if on EFI";
+      type = types.bool;
+      description = ''
+        Whether or not to install the limine EFI files.
+      '';
+    };
+
+    installBios = mkOption {
+      default = pkgs.stdenv.hostPlatform.isi686;
+      defaultText = literalExpression "true if on i686";
+      type = types.bool;
+      description = ''
+        Whether or not to install limine for BIOS.
+      '';
+    };
+
+    biosDevice = mkOption {
+      default = "nodev";
+      type = types.str;
+      description = ''
+        Device to install the BIOS version of limine on.
+      '';
+    };
+
+    partitionIndex = mkOption {
+      default = null;
+      type = types.nullOr types.int;
+      description = ''
+        The 1-based index of the dedicated partition for limine's second stage.
+      '';
+    };
+
+    enrollConfig = mkOption {
+      default = cfg.panicOnChecksumMismatch;
+      defaultText = literalExpression "boot.loader.limine.panicOnChecksumMismatch";
+      type = types.bool;
+      description = ''
+        Whether or not to enroll the config.
+        Only works on EFI!
+      '';
+    };
+
+    forceMbr = mkOption {
+      default = false;
+      type = types.bool;
+      description = ''
+        Force MBR detection to work even if the safety checks fail, use absolutely only if necessary!
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    assertions = [
+      {
+        assertion = pkgs.stdenv.hostPlatform.isx86 || pkgs.stdenv.hostPlatform.isAarch64;
```

Comment:
Consider asserting isi686 here, Limine only supports i686 and later. Maybe this should be asserted in the limine package too?